src/main/java/com/gmail/nossr50/util/EventUtils.java

```diff
@@ -379,6 +379,9 @@ public static boolean handlePartyLevelChangeEvent(Party party, int levelsChanged
     }
 
     public static boolean handleXpGainEvent(Player player, PrimarySkillType skill, float xpGained, XPGainReason xpGainReason) {
+        if(UserManager.getPlayer(player) == null)
```

Comment:
I see `UserManager.getPlayer(player)` is called twice further below, even though it is conditional.
I reckon it would be much more efficient to store it as a variable and use that, to avoid unnecessary calls.
Because with your change, this will be called everytime unconditionally anyway.

Reply:
Assuming the call rarely comes up null, I agree. Was just a quick c/p from the handleStatsLossEvent below. Will make the change once able to do so.